falcon/response.py

```diff
@@ -873,7 +873,9 @@ def _wsgi_headers(self, media_type=None, py2=PY2):
         """
 
         headers = self._headers
-        self._set_media_type(media_type)
+        # PERF(vytas): uglier inline version of Response._set_media_type
+        if media_type is not None and 'content-type' not in self._headers:
+            self.set_header('content-type', media_type)
```

Comment:
Setting self._headers directly would offer an additional performance boost. I think ``_set_media_type()`` was the only place that called ``self.set_header().``

Reply:
Thanks, addressed!